### PR TITLE
Extract dmd.mtype.Type.isBaseOf and move it to dmd.typesem

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

**Comment by RazvanN7** [comment]
@ibuclaw @kinke I made this function `extern(D)` as the cxx tester does not use it. Please confirm that that is indeed the case on the GDC/LDC side.

File: compiler/src/dmd/typesem.d [hunk 1]
```diff
@@ -5658,6 +5658,27 @@ StorageClass parameterStorageClass(TypeFunction tf, Type tthis, Parameter p, Var
         return stc | STC.scope_;
 }
 
+bool isBaseOf(Type tthis, Type t, int* poffset)
```
Comment (thewilsonator) [inline compiler/src/dmd/typesem.d:5661]: this is used from C++

**Review by thewilsonator — CHANGES_REQUESTED** [review]
see above comment

**Comment by RazvanN7** [comment]
@thewilsonator done.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**